### PR TITLE
[13.0][MIG] account_invoice_line_description

### PR DESCRIPTION
**Pull request by kongrattapong**
Ready to review.

### CODE REVIEW

**Review by kittiu — APPROVED** [review]
Functional tested, LGTM :+1:

**Review by SimoneVagile — APPROVED** [review]

**Comment by SimoneVagile** [comment]
This can be merged?

**Review by HaraldPanten — APPROVED** [review]
LGTM

**Review by anasuarez1 — APPROVED** [review]
Functional review LGTM!

Since everything looks ok can we get this merged?

File: account_invoice_line_description/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2015 Agile Business Group sagl (https://www.agilebg.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+{
+    "name": "Account invoice line description",
+    "version": "13.0.1.0.0",
+    "category": "Generic Modules/Accounting",
+    "author": "Agile Business Group, " "Tecnativa, " "Odoo Community Association (OCA)",
```
Comment (ernestotejeda) [inline account_invoice_line_description/__manifest__.py:8]:
```suggestion
    "author": "Agile Business Group, Tecnativa, Odoo Community Association (OCA)",
```

File: account_invoice_line_description/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2015 Agile Business Group sagl (https://www.agilebg.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import api, models
+
+
+class AccountMove(models.Model):
```
Comment (ernestotejeda) [inline account_invoice_line_description/models/account_move.py:7]:
```suggestion
class AccountMoveLine(models.Model):
```

File: account_invoice_line_description/models/account_move.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    _inherit = "account.move.line"
+
+    @api.onchange("product_id")
+    def _onchange_product_id(self):
+        res = super(AccountMove, self)._onchange_product_id()
```
Comment (ernestotejeda) [inline account_invoice_line_description/models/account_move.py:12]:
```suggestion
        res = super()._onchange_product_id()
```

File: account_invoice_line_description/models/account_move.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+            "account_invoice_line_description."
+            "group_use_product_description_per_inv_line"
+        ):
+            inv_type = self.move_id.type
+            product = self.product_id.with_context(lang=self.move_id.partner_id.lang)
+            self.name = (
+                product.description_purchase or self.name
+                if inv_type in ("in_invoice", "in_refund")
+                else product.description_sale or self.name
+            )
```
Comment (ernestotejeda) [inline account_invoice_line_description/models/account_move.py:23]:
```suggestion
            product = self.product_id.with_context(lang=self.move_id.partner_id.lang)
            if self.is_purchase_document():
                self.name = product.description_purchase or self.name
            elif self.is_sale_document():
                self.name = product.description_sale or self.name
```
In this way, the name of the account.move of type 'entry' is not changed.

File: account_invoice_line_description/wizards/res_config.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    group_use_product_description_per_inv_line = fields.Boolean(
+        string="Allow using only the product description on the invoice lines",
```
Comment (ernestotejeda) [inline account_invoice_line_description/wizards/res_config.py:11]: I think we should change this string value, it's to long IMO and is the same as the help.

**Review by ernestotejeda — CHANGES_REQUESTED** [review]
Hello @kongrattapong , thanks for the migration. 
During the merge there was an error, you must rebase and run the pre-commit again.
I also have some change suggestions here, check it out, please ;). 
If you can't continue with the PR, just let me know and I'll open a new one :), thanks.

**Comment by pedrobaeza** [comment]
Superseded by #1077

**Pull request closed without merging**